yarn-project/circuit-types/src/tx/simulated_tx.ts

```diff
@@ -45,6 +48,29 @@ function processOutputFromJSON(json: any): ProcessOutput {
 export class SimulatedTx {
   constructor(public tx: Tx, public privateReturnValues?: ProcessReturnValues, public publicOutput?: ProcessOutput) {}
 
+  /**
+   * Returns suggested total and teardown gas limits for the simulated tx.
+   * Note that public gas usage is only accounted for if the publicOutput is present.
+   * @param pad - Percentage to pad the suggested gas limits by, defaults to 10%.
+   */
+  public getGasLimits(pad = 0.1) {
+    const privateGasUsed = this.tx.data.publicInputs.end.gasUsed;
+    if (this.publicOutput) {
+      const publicGasUsed = Object.values(this.publicOutput.gasUsed).reduce(
+        (total, current) => total.add(current),
+        Gas.empty(),
+      );
+      const teardownGas = this.publicOutput.gasUsed[PublicKernelType.TEARDOWN] ?? Gas.empty();
+
+      return {
+        totalGas: privateGasUsed.add(publicGasUsed).mul(1 + pad),
+        teardownGas: teardownGas.mul(1 + pad),
+      };
+    }
+
+    return { totalGas: privateGasUsed.mul(1 + pad), teardownGas: Gas.empty() };
```

Comment:
Would it be more clear to have this return `Record<PublicKernelType, Gas>` ?

Reply:
Also, is it possible to check this in private_payments.test.ts ? It would be neat to confirm the values match the transaction fee.

Reply:
> Would it be more clear to have this return Record<PublicKernelType, Gas> ?

Makes sense, especially if we also move the scaling out of this.

> Also, is it possible to check this in private_payments.test.ts ? It would be neat to confirm the values match the transaction fee.

Yep, my plan was to get this into the e2e tests as soon as I integrated it with aztecjs in the next PR.